modules/Bio/EnsEMBL/Compara/RunnableDB/FTPDumps/SymlinkPreviousDumps.pm

```diff
@@ -31,63 +27,59 @@ use warnings;
 use strict;
 use Bio::EnsEMBL::Registry;
 use Bio::EnsEMBL::Compara::DBSQL::DBAdaptor;
-use Data::Dumper;
 
 use base ('Bio::EnsEMBL::Compara::RunnableDB::BaseRunnable');
 
-# sub param_defaults {
-#     my ($self) = @_;
-#     return {
-#         %{$self->SUPER::param_defaults},
-#         'param'   => undef,
-#     }
-# }
 
 sub fetch_input {
 	my $self = shift;
 
 	my $ftp_root = $self->param_required('ftp_root');
 	my $dump_dir = $self->param_required('dump_dir');
-	my $curr_release = $self->param_required('curr_release');
-	my $prev_release = $curr_release-1;
+    my $division = $self->param_required('division');
+
+    my ($prev_rel_ftp_root, @cmds, %missing_dump_mlsses);
 
-	my (@cmds, %missing_dump_mlsses);
+    if ($division eq 'vertebrates') {
+        my $prev_release = $self->param_required('curr_release') - 1;
+        $prev_rel_ftp_root = "$ftp_root/release-$prev_release";
+    } else {
+        my $prev_eg_release = $self->param_required('curr_eg_release') - 1;
+        $prev_rel_ftp_root = "$ftp_root/release-$prev_eg_release/$division";
```

Comment:
This feels like a well used variable - is there a way it could be passed in as a parameter through the pipe_conf?

Reply:
This is the only solution I have been able to come with that works:
```
sub pipeline_wide_parameters {
    return {
        ...
        'prev_rel_ftp_root' => '#expr( #ftp_root# . "/release-" . ((#division# eq "vertebrates") ? #curr_release# - 1 : (#curr_eg_release# - 1) . "/" . #division#) . ((#division# eq "pan") ? "_ensembl" : "") )expr#',
...
```
It is a horrible expression, I know :sweat_smile: What do you think?

Reply:
I don't think it is too horrible, although a long line. I was also thinking it something that could be default value for verts and put in the division specific pipe_confs for each division?